### PR TITLE
Dart - Shorten URL function

### PR DESCRIPTION
**Pull request by letsintegreat**
This PR adds an example of Appwrite function to shorten URLs using bitly and tinyurl written in dart.

Closes https://github.com/appwrite/appwrite/issues/4184

I was not able to implement owly service, because it seems its API is not available for free usage, as mentioned here in the docs:

image

The steps to deploy the function and test it are provided in the README.md file. 

Working screenshots of the function:

1. bitly success

bitly_success

2. tinyurl success

tinyurl_success

3. Invalid URL

tinyurl_failure

4. Invalid API key

bitly_failure

### CODE REVIEW

File: dart/generate_short_url/test.py
Comment (stnguyen90) [inline dart/generate_short_url/test.py:53]: It seems odd to have a python script in a dart example. Can this be a dart script instead?

File: dart/generate_short_url/lib/utils/api.dart
Comment (stnguyen90) [inline dart/generate_short_url/lib/utils/api.dart:7]: Would you please ensure configurable values are read through `req.variables` rather than being hardcoded in the code? For example:

https://github.com/appwrite/functions-starter/blob/53948b924db3bab390e03c4d10970add1b0de8bb/dart-2.17/lib/main.dart#L37

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: dart/generate_short_url/lib/utils/api.dart
Comment (letsintegreat) [reply dart/generate_short_url/lib/utils/api.dart:7]: @stnguyen90 I need some help, I can't seem to be able to set environment variables. How do we set env variables so that they can be accessed using req.variables?

I tried putting them in `.env` file of appwrite, and restarted it. But no luck.

Comment (stnguyen90) [reply dart/generate_short_url/lib/utils/api.dart:7]: They're not really environment variables. They're passed in as part of the request.

Comment (letsintegreat) [reply dart/generate_short_url/lib/utils/api.dart:7]: Ah, do you mean the variables appended to the URL after a `?`?

For eg - `localhost:3000/?apiKey=somekey`

Comment (letsintegreat) [reply dart/generate_short_url/lib/utils/api.dart]: @stnguyen90 I have updated the function to now take configurable values through `req.variables`. PTAL!

File: dart/generate_short_url/test.py
Comment (letsintegreat) [reply dart/generate_short_url/test.py]: @stnguyen90 Because now API Key is being sent through the request, I have updated the PR to now have screenshots of Postman instead, for more charity. Kindly take a look at those screenshots to verify that the function is working properly instead of running the python script (Now removed).

File: dart/generate_short_url/README.md
Comment (stnguyen90) [inline dart/generate_short_url/README.md:1]:
```suggestion
# 🌐 Shorten any URL using bitly and tinyurl.
```

File: dart/generate_short_url/lib/main.dart
Comment (stnguyen90) [inline dart/generate_short_url/lib/main.dart:101]: Would you please add separate variables for the API keys?

```
BITLY_API_KEY
TINYURL_API_KEY
```

These should then be documented in the README. For example, see [this](https://github.com/open-runtimes/examples/blob/main/node/send_email_with_mailgun/README.md) README.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: dart/generate_short_url/README.md
Comment (letsintegreat) [reply dart/generate_short_url/README.md]: Done!

File: dart/generate_short_url/lib/main.dart
Comment (letsintegreat) [reply dart/generate_short_url/lib/main.dart]: Ah, totally forgot updating the README, thanks! I have now separated the variables for different providers, and updated the info in README.

File: dart/generate_short_url/README.md
Comment (Meldiron) [inline dart/generate_short_url/README.md:5]: Let's please add one more example input where provider is `tinyurl`, to make it clear what exact words needs to be passed in.

**Review by Meldiron — CHANGES_REQUESTED** [review]
I love this PR! 😍 Amazing quality of both PR itself and the code too. Leaving one small comment. Once it is resolved, I can mark this as approved 👍

File: dart/generate_short_url/README.md
Comment (letsintegreat) [reply dart/generate_short_url/README.md]: @Meldiron Thanks for reviewing, I have addressed this requested change.

**Review by Meldiron — APPROVED** [review]

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**
